synapse_invite_checker/rest/contacts.py

```
@@ -60,18 +64,29 @@ async def on_GET(self, _: SynapseRequest) -> tuple[int, JsonDict]:
 
 
 class ContactsResource(RestServlet):
-    def __init__(self, api: ModuleApi, store: InviteCheckerStore):
+    def __init__(
+        self,
+        api: ModuleApi,
+        store: InviteCheckerStore,
+        permission_handler: InviteCheckerPermissionsHandler,
+    ):
         super().__init__()
         self.store = store
         self.api = api
+        self.permission_handler = permission_handler
         self.PATTERNS = contact_mgmt_patterns("/contacts$")
 
-    # @override
     async def on_GET(self, request: SynapseRequest) -> tuple[int, JsonDict]:
         requester = await self.api.get_user_by_req(request)
+        await self.permission_handler.do_migration(self.store, requester.user)
+
+        contacts = await self.permission_handler.get_contacts(
```

I think this should return an error if the user flipped the allow_all bit, because then this API shouldn't be used anymore imo?

---

I had made the assumption that the client app would be allowed time to roll out some of the changes and had considered the likeliest option was migration first and *then* allowing that to change. From the moment of migration the data is available to the client and it shouldn't need to use the contact API at all

---

What error code do you think would be appropriate? 403?